keras/applications/imagenet_utils.py

```diff
@@ -89,17 +90,26 @@ def _obtain_input_shape(input_shape,
         data_format: image data format to use.
         include_top: whether the model is expected to
             be linked to a classifier via a Flatten layer.
+        weights: one of `None` (random initialization)
+            or "imagenet" (pre-training on ImageNet).
+            If weights="imagenet" input channels must be equal to 3.
 
     # Returns
         An integer shape tuple (may include None entries).
 
     # Raises
         ValueError: in case of invalid argument values.
     """
-    if data_format == 'channels_first':
-        default_shape = (3, default_size, default_size)
+    if weights != 'imagenet' and input_shape is not None and len(input_shape) == 3:
```

Comment:
If the channel dimension isn't 1 or 3, display a warning (because it's very likely to be a user error in that case). We have a similar warning in `ImageDataGenerator`.